File: 2553-separate-the-digits-in-an-array/2553-separate-the-digits-in-an-array.cpp
class Solution {
public:
    vector<int> separateDigits(vector<int>& nums) {
        vector<int> digits;

        for(int i=nums.size()-1 ; i>=0 ; i--) {
            int n = nums[i];
            while(n!=0) {
                int digit = n % 10;
                digits.push_back(digit);
                n = n / 10;
            }
        }
        int n = digits.size();
        vector<int> result(n);

        for(int i=0; i<n ; i++) {
            result[i] = digits[n-i-1];
        }
        return result;
    }
};

File: 2553-separate-the-digits-in-an-array/2553-separate-the-digits-in-an-array.py
class Solution(object):
    def separateDigits(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        digits = []

        for i in range(len(nums) - 1, -1, -1):
            n = nums[i]
            while n != 0:
                digit = n % 10
                digits.append(digit)
                n = n // 10  

        n_size = len(digits)
        result = [0] * n_size 

        for i in range(n_size):
            result[i] = digits[n_size - i - 1]
            
        return result